keps/sig-apps/3088-pdb-for-daemonset-pods/README.md

```diff
@@ -0,0 +1,252 @@
+  controllers.
+- Reduce the load to kube-apiserver caused by the recreation of DaemonSet pods being evicted.
+
+### Non-Goals
+
+- Introduce API changes.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+For now, PDB does not support DaemonSet pods. This KEP aims to add this support.
+
+### User Stories
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+In a cluster with big number of nodes, DaemonSet pods can be evicted by VPA. The load of re-creation
+of pods is scaled as number of nodes multiplied by number of DaemonSets being affected
+simultaneously. In order to reduce the load, one can set a PDB for DaemonSet pods.
+
+For example, in case of 100 nodes and 3 DaemonSets affected by an eviction, one would prefer the
+simultaneous recreation of 30 pods instead of 300.
+
+From the API perspective, PDB status for DaemonSet will contain calculated distuprion-related
+numbers, as for any other kubernetes controller.
+
+<!-- ### Notes/Constraints/Caveats (Optional) -->
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+<!-- ### Risks and Mitigations -->
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+In disruption controller, DaemonSet pods can be supported the same way as for other built-in pods
+controllers. It is easily achieved single the code is quite similar for all controllers. The
+specific part for DaemonSets is the calculation of allowed disruption from its status.
+
+### Test Plan
+
+In practice, the patch provided by PR [k/k #98307](https://github.com/kubernetes/kubernetes/pull/98307)
+proves it works in production environments at Flant since March 2020
+([kube-controller-manager patch in deckhouse/deckhouse](https://github.com/deckhouse/deckhouse/blob/main/modules/040-control-plane-manager/images/kube-controller-manager/patches/1.21/pdb-daemonset.patch)
+is used in 1.18, 1.19, 1.20, 1.21).
```

Comment:
This isn't convincing enough test plan - we need test coverage, not a single working environment.
[It's not only a question if it works now - it's even more if it won't get broken in the future.]

Reply:
Agree, mentioned unit test since it's how the disruption controller is tested.